python/tank/pipelineconfig_factory.py

```diff
@@ -381,6 +387,19 @@ def _validate_and_create_pipeline_configuration(associated_pipeline_configs, sou
                     "definition in Shotgun." % (sg_config_data["id"], source)
                 )
 
+            # This checks for a very subtle bug. If the toolkit_init.cache contains a path to a pipeline
+            # configuration that matches the pre-requisites, but that doesn't actually exist on disk because
+            # it was either move to another location or delete from disk altogether, then we need to raise
+            # TankInitError. If the cache hadn't been force reread, this will be caught by the factory
+            # and Shotgun will be queried once again for the pipeline configuration info, hopefully
```

Comment:
This is the edge case I was mentioning.

Reply:
nitpick: `moved to another location or deleted`